### PR TITLE
feat: Add auto complete component with comboBox and freeSolo modes

### PR DESCRIPTION
**Pull request by PuBHARGAVI**
- Created Auto Complete component which supports comboBox and freeSolo modes and implemented different scenarios for both the modes taking material UI as reference
- Added test cases for all the scenarios

### CODE REVIEW

**Comment by vczb** [comment]
hey! thanks for your PR, please try to fix the build running `yarn prettier:format`

File: src/packages/Tag/styles.ts [hunk 1]
```diff
@@ -17,18 +17,20 @@ const wrapperModifier = {
...
     isBold,
     isPointer,
-    size = 'small'
+    size = 'small',
+    hovered,
+    isSelected
   }) => css`
     padding: ${theme.spacings.mini} ${theme.spacings.xxsmall};
-    background: ${theme.colors.primary.light};
+    background: ${isSelected ? theme.colors.secondary.dark : theme.colors.primary.light};
```
Comment (vczb) [inline src/packages/Tag/styles.ts:33]: let's make it as generic as possible allowing it to make sense in different situations, so to achieve it I would like to suggest renaming `isSelected` for `isDarkBg`. By doing this we can set this prop in many use-cases in the future

File: src/packages/Tag/styles.ts [hunk 2]
```diff
@@ -39,5 +41,8 @@ export const Wrapper = styled.div<
     ${!!isPointer && wrapperModifier.isPointer()};
     ${!!isSlashed && wrapperModifier.isSlashed()};
     ${!!isBold && wrapperModifier.isBold(theme)};
+    &:hover {
+      background: ${hovered ? theme.colors.secondary.dark : theme.colors.primary.light}
```
Comment (vczb) [inline src/packages/Tag/styles.ts:45]: same situation here, instead of `hovered` let's rename for `isDarkBgOnHover`

**Review by vczb — COMMENTED** [review]
hey! I really like your PR. I left some suggestions for the props names on Tag Component, applying it I can approve and merge :)

File: src/packages/Tag/index.tsx [hunk 3]
```diff
@@ -8,6 +8,8 @@ export type TagProps = {
   isPointer?: boolean
   isSlashed?: boolean
   size?: 'xsmall' | 'small' | 'medium'
+  hovered?: boolean
```
Comment (vczb) [inline src/packages/Tag/index.tsx:11]: I left some suggestions for renaming both props

File: src/packages/Tag/styles.ts (hunk 1 above)
Comment (PuBHARGAVI) [reply src/packages/Tag/styles.ts]: @vczb Thanks for the suggestions. I have made the changes please look into it and let me know if i have to make any other changes
Done

**Review by vczb — APPROVED** [review]
LGTM

File: src/packages/Tag/styles.ts (hunk 2 above)
Comment (PuBHARGAVI) [reply src/packages/Tag/styles.ts]: Done

File: src/packages/Tag/index.tsx (hunk 3 above)
Comment (PuBHARGAVI) [reply src/packages/Tag/index.tsx]: Done

**Pull request merged**